py/kubeflow/testing/util.py

```diff
@@ -93,6 +99,34 @@ def run(command,
 def run_and_output(*args, **argv):
   return run(*args, **argv)
 
+def build_repos_dict(main_repo, extra_repos):
```

Comment:
Where does main_repo come from? Is it coming from the prow env?

Would it be useful to have a function

```
def prow_var_to_repo_spec(prow_dict)
```

That would take the prow variables and convert it to the repo spec? I find myself writing that code over and over again.  e.g.

https://github.com/kubeflow/kubeflow/blob/41dd8167ff4171d6acf12822d19c571230cf04ce/py/kubeflow/kubeflow/ci/kfctl_go_test_utils.py#L189

Would it be simpler to simplify the function signature to just be

```
def combine_repos(list_of_repos)
```

list_of_repos would then just be a list of repos in the form ${ORG}/{REPO}:version

Combine would then group the repos by {ORG}/{REPO} with the value on the right taking precedence.

So then in the case of kubeflow/kubeflow we could just write

combine_repos(["kubeflow/kubeflow@HEAD", "kubeflow/tf_operator@HEAD", prow_var_to_repo_spec(os.environ)] + extra_repos)

Reply:
In the latter example, does it mean that the caller is expected to pass in default repos to combine_repos? That would require us to modify more places.

Reply:
Also, there is https://github.com/kubeflow/testing/blob/master/py/kubeflow/testing/argo_build_util.py#L211

So outside of this method, we could do something like
```
repos = combine_repos([argo_build_util.get_repo_from_prow_env()] + extra_repos)
repos_str = ','.join(['%s@%s' % (key, value) for (key, value) in repos.items()])
```

Reply:
> In the latter example, does it mean that the caller is expected to pass in default repos to combine_repos? That would require us to modify more places.

Doesn't the caller have to pass in default repos? The default repos are test specific. e.g.

```
DEFAULT_REPOS = {
    "kubeflow/kubeflow": "HEAD",
    "kubeflow/testing": "HEAD",
    "kubeflow/tf-operator": "HEAD"
}
```

These defaults are specific to the kfctl_e2e_test. But they wouldn't make sense as the defaults for other tests; e.g. some of the E2E tests in kubeflow/examples